agent/grpc-external/services/peerstream/stream_resources.go

```diff
@@ -161,8 +161,19 @@ func (s *Server) StreamResources(stream pbpeerstream.PeerStreamService_StreamRes
 	if p == nil {
 		return grpcstatus.Error(codes.InvalidArgument, "initial subscription for unknown PeerID: "+req.PeerID)
 	}
+	if !p.IsActive() {
+		logger.Warn("peering is marked as deleted or terminated", "peer_id", req.PeerID)
+		term := &pbpeerstream.ReplicationMessage{
+			Payload: &pbpeerstream.ReplicationMessage_Terminated_{
+				Terminated: &pbpeerstream.ReplicationMessage_Terminated{},
+			},
+		}
+		logTraceSend(logger, term)
 
-	// TODO(peering): If the peering is marked as deleted, send a Terminated message and return
+		// we don't care if send fails; stream will be killed by termination message or grpc error
+		_ = stream.Send(term)
+		return grpcstatus.Error(codes.FailedPrecondition, "peering is marked as deleted: "+req.PeerID)
```

Comment:
Hmm I just noticed that this is another use-case for `codes.FailedPrecondition`. This code is one that can trigger a tight loop by the dialer on the assumption that the precondition is that we dialed a follower. But in this case that's not true.

I'll write up an issue about that.

Reply:
Yeah I think we need a pass through our codes and try to make them as standardized as possible. I could make this another code to prevent that tight loop